frontend/webservice/vrf.cgi

```diff
@@ -0,0 +1,307 @@
+        pattern         => $GRNOC::WebService::Regex::INTEGER,
+        required        => 0,
+        description     => "If editing an existing VRF specify the ID otherwise leave blank for new VRF."
+        );
+
+    $method->add_input_parameter(
+        name            => 'name',
+        pattern         => $GRNOC::WebService::Regex::NAME_ID,
+        required        => 1,
+        description     => "The workgroup_id with permission to build the vrf, the user must be a member of this workgroup."
+        );
+
+    $method->add_input_parameter(
+        name            => 'workgroup_id',
+        pattern         => $GRNOC::WebService::Regex::INTEGER,
+        required        => 1,
+        description     => "The workgroup_id with permission to build the vrf, the user must be a member of this workgroup."
+        );
+
+    #add the required input parameter description
+    $method->add_input_parameter(
+        name            => 'description',
+        pattern         => $GRNOC::WebService::Regex::TEXT,
+        required        => 1,
+        description     => "The description of the circuit."
+        );    
+    
+    $method->add_input_parameter(
+        name            => 'endpoint',
+        pattern         => $GRNOC::WebService::Regex::TEXT,
+        required        => 1,
+        multiple        => 1,
+        description     => "The JSON blob describing all of the endpoints"
+    );
+
+    $method->add_input_parameter(
+        name            => 'local_asn',
+        pattern         => $GRNOC::WebService::Regex::INTEGER,
+        required        => 1,
+        description     => "The JSON blob describing all of the endpoints"
+    );
+
+    $method->add_input_parameter(
+        name            => 'prefix_limit',
+        pattern         => $GRNOC::WebService::Regex::INTEGER,
+        required        => 0,
+        default         => 1000,
+        description     => "Maximum prefix limit size for BGP peer routes"
+    );
+
+    $svc->register_method($method);
+    
+    $method = GRNOC::WebService::Method->new(
+        name => 'remove',
+        description => 'removes a vrf that is on the network',
+        callback => sub { remove_vrf(@_) });    
+    
+    $method->add_input_parameter(
+        name            => 'workgroup_id',
+        pattern         => $GRNOC::WebService::Regex::INTEGER,
+        required        => 1,
+        description     => "The workgroup_id with permission to build the vrf, the user must be a member of this workgroup."
+        );
+
+    $method->add_input_parameter(
+        name => 'vrf_id',
+        pattern => $GRNOC::WebService::Regex::INTEGER,
+        required => 1,
+        description => 'the ID of the VRF to remove from the network');
+
+    $svc->register_method($method);
+
+}
+
+sub get_vrf_details{
+    my $method = shift;
+    my $params = shift;
+    my $ref = shift;
+
+    my $vrf_id = $params->{'vrf_id'}{'value'};
+    
+    my $vrf = OESS::VRF->new(db => $db, vrf_id => $vrf_id);
```

Comment:
Check that vrf is valid before calling `$vrf->to_hash`

Reply:
done